content/6_Advanced/XOR_Basis.mdx

```diff
@@ -25,8 +24,152 @@ frequency: 1
 	>
 		used at USACO Camp
 	</Resource>
+	<Resource
+		source="hackmd"
+		url="https://hackmd.io/@minhnguyent546/BJKcH_2ti#XOR-BASIS"
+		title="Minh Thien Nguyen - XOR BASIS"
+		starred
+	>
+		detailed explanation about XOR basis note: translate the page to english
+	</Resource>
 </Resources>
 
+<FocusProblem problem="xorqueries" />
+
+You are given a tree consisting of $n$ vertices. There is an integer written on each vertex; the $i$-th vertex has integer $$a_i$$ written on it. You have to process $q$ queries. The $i$-th query consists of three integers $$x_i$$, $$y_i$$ and $$k_i$$. For this query, you have to answer if it is possible to choose a set of vertices $v_1,v_2,…,v_m$ (possibly empty) such that:
+- every vertex $$v_j$$ is on the simple path between $$x_i$$ and $$y_i$$ (endpoints can be used as well);
+- $$a_{v_1} \oplus a_{v_2} \oplus \dots \oplus a_{v_m} = k_i$$, where $\oplus$ denotes the bitwise XOR operator.
+
+## Solution - Trees and XOR Queries Again
+
+<!-- [Official Tutorial](https://codeforces.com/blog/entry/122951) -->
+
+To efficiently compute XOR bases on paths in a tree, we use a method involving tree rooting, **Lowest Common Ancestor (LCA)**, and properties of **XOR bases**. The process begins by rooting the tree and using LCA to split any path into two vertical paths. For each vertex $$v$$, we maintain a list of "interesting" vertices that significantly influence the XOR base when traversing from $$v$$ to the root. Due to the properties of XOR bases, these lists are small, with a maximum size of 20.
```

Comment:
> To efficiently compute XOR bases

ok but this module is about...the xor basis
i would suggest including a problem introducing the concept in the first place

Reply:
XOR basis is like one of the rarest topics, to get some content on this topic I find it very difficult as there is nothing much about it on the web or even in books. If you could provide a problem then that would be beneficial.

Reply:
idk i googled and the first result that came up was this: https://www.geeksforgeeks.org/xor-basis-algorithm/#

Reply:
I had already read this blog, I just don't find it a good blog. Anyway I will try to add an algorithm section to it along with a question.

Reply:
Made all the changes and added new content to it. please take a look at it.